pkg/blockstore/shreds.go

```diff
@@ -97,10 +103,33 @@ func DataShredsToEntries(meta *SlotMeta, shreds []shred.Shred) (entries []Entrie
 			Raw:     entryBytes[:dec.Position()],
 			Shreds:  parts,
 		})
+		decoderPool.Put(dec)
 	}
 	return entries, nil
 }
 
+type SubEntries struct {
+	NumEntries uint64
+	Entries    []shred.Entry
+}
+
+func (se *SubEntries) UnmarshalWithDecoder(decoder *bin.Decoder) (err error) {
+	// read the number of entries:
+	numEntries, err := decoder.ReadUint64(bin.LE)
+	if err != nil {
+		return fmt.Errorf("failed to read number of entries: %w", err)
+	}
+	se.NumEntries = uint64(numEntries)
+	// read the entries:
+	se.Entries = make([]shred.Entry, se.NumEntries)
```

Comment:
unsafe slice creation (length from user-provided input)
Some sort of bounds check would be nice

Reply:
Added a check against the number of bytes remaining.